code/game/objects/items/devices/dna_scanner.dm

```diff
@@ -0,0 +1,110 @@
+/obj/item/device/dna_scanner
+	name = "dna analyzer"
+	desc = "A hand-held scanner able to distinguish mutations in DNA of the subject injected with kognim."
+	icon_state = "dna"
+	item_state = "analyzer"
+	throw_speed = 5
+	throw_range = 10
+	var/use_delay = 170
+	var/charge_per_use = 0
+	var/obj/item/weapon/cell/cell = null
+	var/suitable_cell = /obj/item/weapon/cell/small
+	var/list/scanned_mutations = list()
+	var/scan_sound = 'sound/machines/twobeep.ogg'
+	matter = list(MATERIAL_PLASTIC = 2, MATERIAL_GLASS = 1)
+	origin_tech = list(TECH_MAGNET = 1, TECH_BIO = 2)
+
+obj/item/device/dna_scanner/proc/can_use(mob/user)
+	if (user.incapacitated())
+		return
+	if (!user.IsAdvancedToolUser())
+		return
+	if(!cell_use_check(charge_per_use))
+		to_chat(user, SPAN_WARNING("[src] battery is dead or missing."))
+		return
+	return TRUE
+
+/obj/item/device/dna_scanner/afterattack(atom/A, mob/user, proximity)
+	if(!proximity)
+		return
+	if(!can_use(user))
+		return
+
+	if(istype(A, /obj/item/modular_computer))
+		to_chat(user, "Transfering scanned mutations to \the [A].")
+		var/obj/item/modular_computer/MC = A
+		if(!MC.enabled)
+			to_chat(user, "You cannot transfer any results to \the turned off [A].")
+		else if (!MC.dna_processor)
+			to_chat(user, "You cannot transfer any results to \the [A] without DNA processor.")
+		else
+			MC.dna_processor.addMutations(scanned_mutations)
+	else
+		if(is_valid_scan_target(A) && A.simulated)
+			user.visible_message(SPAN_NOTICE("[user] runs \the [src] over \the [A]."), range = 2)
+			if(scan_sound)
+				playsound(src, scan_sound, 30)
+			if(use_delay && !do_after(user, use_delay, A))
+				to_chat(user, "You stop scanning \the [A] with \the [src].")
+				return
+			scan(A, user)
+		else
+			to_chat(user, "You cannot get any results from \the [A] with \the [src].")
+
+/obj/item/device/dna_scanner/proc/cell_check()
+	if(!cell && suitable_cell)
+		cell = new suitable_cell(src)
```

Comment:
Why make an entire proc and not keep it in `Initialize`, like this is done everywhere else?